dart/lib/src/sentry_client.dart

```diff
@@ -52,6 +52,7 @@ class SentryClient {
     dynamic stackTrace,
     dynamic hint,
   }) async {
+    event = event.copyWith(); // clone with mutable list and map fields
```

Comment:
guess I already asked it, but just to double-check, if we are defending ourselves now and copying all the List/Map references, in which cases do we run into immutable problems? do we still need this?

Reply:
✅ true ! I updated the SentryEvent ctor and copyWith to avoid the need for this copyWith call